### ISSUE TITLE
Problem using rabbitmq transaction

### ISSUE THREAD

**Issue by mboutet** [cloudamqp/amqproxy#39]
Hi, I'm using RabbitMQ v3.8.9 with AMQProxy v0.4.7 and it seems that AMQProxy can't handle transactions.

Here is a portion of AMQProxy's logs showing the error when one of my service tries to use a transaction:
```
10 clients		 16 upstreams
11 clients		 16 upstreams
Upstream closed connection: FRAME_ERROR - cannot decode <<>>
2020-10-16T18:27:13.270997652Z 
Upstream connection closed when returned
2020-10-16T18:28:19.077851087Z 
10 clients		 15 upstreams
9 clients		 15 upstreams
```

When the service is directly connected to the broker, transactions are working fine.

Is this a known limitation/bug of AMQProxy? If so, is this something that can be supported in future versions? If I can provide any additional informations, let me know.

Thank you in advance.

Just a bit of context why I use transaction in this service: I know that transactions are generally not recommended since they can significantly decrease the throughput (according to [this](https://www.rabbitmq.com/confirms.html#publisher-confirms)). However, I specifically need them in this service because I'm sending a large amount of messages (few thousands) on a work queue. If I don't use a transaction, the consumers will start sending processed messages back even though the publishing service is not done dispatching the messages, thus causing weird race conditions.

**Comment by carlhoerberg** [comment]
Hi! Yes, this should be fairly simple, just `shards update && shards build --release` as transaction support is now available in amq-protocol.cr

**Comment by carlhoerberg** [comment]
0.4.8 is out, at https://packagecloud.io/cloudamqp/amqproxy or https://github.com/cloudamqp/amqproxy/actions/runs/317310641

**Comment by mboutet** [comment]
Hi, it seems that the latest version still has the same problem

**Comment by johanrhodin** [comment]
OK, can you provide a minimal example showing the issue?

**Comment by mboutet** [comment]
Sure, I will try to come up with a minimal example. I will poke you when it's ready.

**Comment by mboutet** [comment]
@johanrhodin, I created a minimal example at https://github.com/mboutet/amqproxy/tree/issue-39-example.

1. `cd issue-39`
2. `docker-compose build` (you only have to do this once)
3. `docker-compose up`

You should see `FRAME_ERROR` in the logs. You can uncomment `producer-rabbitmq` in the `docker-compose.yml` and see that connecting directly to rabbitmq works.

**Comment by carlhoerberg** [comment]
Thank you! 

Fixed in master and v0.5.0
